pyvista/core/dataset.py

```diff
@@ -47,6 +47,8 @@
     from collections.abc import Generator
     from collections.abc import Iterator
 
+    import scipy
+
```

Comment:
Since `scipy` is not currently included in our dependencies, we need to check whether it can be imported. We also need to set up a process for when it cannot be imported.
https://github.com/pyvista/pyvista/blob/9a88c38b87969899709289abc0b45c42cfc1e5a0/pyproject.toml#L29C1-L37

Reply:
Good catch this should be for type-checking only, as validate_transform3x3 otherwise will handle the import at runtime. I wonder why ruff didn't suggest moving it to a type-checking block

Reply:
So, it's already in a type checking block. If type-checking anything, the code will still be able to continue type-checking when it hits this line, but missing imports may be imported as `Any`. 

A similar import is already in `main`, merged from #6318:
https://github.com/pyvista/pyvista/blob/9a88c38b87969899709289abc0b45c42cfc1e5a0/pyvista/plotting/prop3d.py#L15-L16

So if this is an issue we should fix it there too

Reply:
Maybe we should add `scipy` as a type-checking dependency first, in a separate PR?

Reply:
> Maybe we should add `scipy` as a type-checking dependency first, in a separate PR?

Yes let's do it. Thanks :)

Reply:
Given that we cannot use scipy in pre-commit mypy, I think that the current state is fine.  We can add in scipy to our type checking once we get it out of pre-commit.

Reply:
This should be resolved now by using the `RotationLike` alias as part of https://github.com/pyvista/pyvista/pull/6354/commits/48aa6e6d335638dba04731bb03d6d9330e080558.

There is still the outstanding issue of not having `SciPy` as a dependency as part of the `mypy` config (see #6368), but that's a separate issue from this PR.